main.nf

```diff
@@ -32,30 +32,37 @@ if (params.aligner != 'bismark' && params.aligner != 'bwameth'){
 }
 if( params.bismark_index && params.aligner == 'bismark' ){
     bismark_index = Channel
-        .fromPath(params.bismark_index)
+        .fromPath(params.bismark_index, checkIfExists: true)
         .ifEmpty { exit 1, "Bismark index not found: ${params.bismark_index}" }
 }
 else if( params.bwa_meth_index && params.aligner == 'bwameth' ){
     bwa_meth_indices = Channel
-        .fromPath( "${params.bwa_meth_index}*" )
+        .fromPath("${params.bwa_meth_index}*", checkIfExists: true)
         .ifEmpty { exit 1, "bwa-meth index not found: ${params.bwa_meth_index}" }
 }
 else if( params.fasta_index && params.aligner == 'bwameth' ){
-    fasta_index = file(params.fasta_index)
-    if( !fasta_index.exists() ) exit 1, "Fasta index file not found: ${params.fasta_index}"
+    fasta_index = Channel
+        .fromPath(params.fasta_index, checkIfExists: true)
 }
 else if( !params.fasta ) {
     exit 1, "No reference genome index specified!"
 }
+
 if ( params.fasta ){
-    fasta = file(params.fasta)
-    if( !fasta.exists() ) exit 1, "Fasta file not found: ${params.fasta}"
+    Channel
+        .fromPath(params.fasta, checkIfExists: true)
+        .into { fasta_1; fasta_2; fasta_3 }
```

Comment:
Can you please use names that are more explicit?

`ch_fasta_for_XYZ` is something that helps you (and me, others) to keep things traceable :-)